File: CS/CustomMapColorizer/ColorGenerator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Media;

namespace CustomMapColorizer
{
    public static class ColorGenerator
    {
        static RangeColor[] colorsPopulation = { new RangeColor(200000, 1400000, Color.FromArgb(255, 0xA8, 0x3A, 0x78)), new RangeColor(34000, 200000, Color.FromArgb(255, 0xBF, 0x51, 0x8F)), new RangeColor(22000, 34000, Color.FromArgb(255, 0xD2, 0x64, 0xA2)),
                                                 new RangeColor(12700, 22000, Color.FromArgb(255, 0xE2, 0x74, 0xB2)), new RangeColor(9200, 12700, Color.FromArgb(255, 0xF1, 0x83, 0xC1)), new RangeColor(6200, 9200, Color.FromArgb(255, 0xFD, 0x8F, 0xCD)),
                                                 new RangeColor(4200, 6200, Color.FromArgb(255, 0xFF, 0xAB, 0xDA)), new RangeColor(2200, 4200, Color.FromArgb(255, 0xFF, 0xC1, 0xE4)), new RangeColor(650, 2200, Color.FromArgb(255, 0xFF, 0xD1, 0xEA)),
                                                 new RangeColor(-100, 650, Color.FromArgb(255, 0xFF, 0xDF, 0xF1))};

        public static Color GetColorByPopulation(int population)
        {
            foreach (RangeColor rangePopulation in colorsPopulation)
            {
                if (population >= rangePopulation.RangeMin * 1000 && population < rangePopulation.RangeMax * 1000)
                {
                    return rangePopulation.Fill;
                }
            }
            return Color.FromArgb(255, 255, 255, 255);
        }
    }
}

File: VB/CustomMapColorizer/ColorGenerator.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows.Media

Namespace CustomMapColorizer
	Public NotInheritable Class ColorGenerator
		Private Shared colorsPopulation() As RangeColor = { New RangeColor(200000, 1400000, Color.FromArgb(255, &HA8, &H3A, &H78)), New RangeColor(34000, 200000, Color.FromArgb(255, &HBF, &H51, &H8F)), New RangeColor(22000, 34000, Color.FromArgb(255, &HD2, &H64, &HA2)), New RangeColor(12700, 22000, Color.FromArgb(255, &HE2, &H74, &HB2)), New RangeColor(9200, 12700, Color.FromArgb(255, &HF1, &H83, &HC1)), New RangeColor(6200, 9200, Color.FromArgb(255, &HFD, &H8F, &HCD)), New RangeColor(4200, 6200, Color.FromArgb(255, &HFF, &HAB, &HDA)), New RangeColor(2200, 4200, Color.FromArgb(255, &HFF, &HC1, &HE4)), New RangeColor(650, 2200, Color.FromArgb(255, &HFF, &HD1, &HEA)), New RangeColor(-100, 650, Color.FromArgb(255, &HFF, &HDF, &HF1))}

		Private Sub New()
		End Sub
		Public Shared Function GetColorByPopulation(ByVal population As Integer) As Color
			For Each rangePopulation As RangeColor In colorsPopulation
				If population >= rangePopulation.RangeMin * 1000 AndAlso population < rangePopulation.RangeMax * 1000 Then
					Return rangePopulation.Fill
				End If
			Next rangePopulation
			Return Color.FromArgb(255, 255, 255, 255)
		End Function
	End Class
End Namespace

